src/test/scala/org/broadinstitute/dsde/firecloud/service/NihApiServiceSpec.scala

```diff
@@ -0,0 +1,93 @@
+    assert(!services.rawlsDao.groups(dbGapAuthorizedUsersGroupName).contains(services.thurloeDao.normalUserId))
+    Post("/nih/callback", validJwtOnWhitelist) ~> dummyUserIdHeaders(services.thurloeDao.normalUserId) ~> sealRoute(services.nihRoutes) ~> check {
+      status should equal(OK)
+    }
+    assert(services.rawlsDao.groups(dbGapAuthorizedUsersGroupName).contains(services.thurloeDao.normalUserId))
+  }
+
+  it should "return InternalServerError when NIH linking with a valid JWT and the user is NOT on the whitelist" in withDefaultApiServices { services =>
+    Post("/nih/callback", validJwtNotOnWhitelist) ~> dummyUserIdHeaders(services.thurloeDao.normalUserId) ~> sealRoute(services.nihRoutes) ~> check {
+      status should equal(InternalServerError)
+    }
+    assert(!services.rawlsDao.groups(dbGapAuthorizedUsersGroupName).contains(services.thurloeDao.normalUserId))
+  }
+
+  it should "return OK when an expired user re-links. their new link time should be 30 days in the future" in withDefaultApiServices { services =>
+    //verify that their link is indeed already expired
+    Get("/nih/status") ~> dummyUserIdHeaders(services.thurloeDao.linkedButExpiredUserId) ~> sealRoute(services.nihRoutes) ~> check {
+      status should equal(OK)
+      assert(responseAs[NihStatus].linkExpireTime.get < DateUtils.now)
+    }
+
+    //link them using a valid JWT for a user on the whitelist
+    Post("/nih/callback", validJwtOnWhitelist) ~> dummyUserIdHeaders(services.thurloeDao.linkedButExpiredUserId) ~> sealRoute(services.nihRoutes) ~> check {
+      status should equal(OK)
+    }
+
+    //verify that their link expiration has been updated
+    Get("/nih/status") ~> dummyUserIdHeaders(services.thurloeDao.linkedButExpiredUserId) ~> sealRoute(services.nihRoutes) ~> check {
+      status should equal(OK)
+      val linkExpireTime = responseAs[NihStatus].linkExpireTime.get
+      //gives a 60 second window to make sure that the link was updated to *approximately* now + 30 days
+      assert(DateUtils.nowPlus30Days - 60000L < linkExpireTime && linkExpireTime <= DateUtils.nowPlus30Days)
+    }
+    assert(services.rawlsDao.groups(dbGapAuthorizedUsersGroupName).contains(services.thurloeDao.linkedButExpiredUserId))
+  }
+
+  /* Test scenario:
+     1 user that is linked but their TCGA access has expired. they should be removed
+     1 user that is linked but they have no expiration date stored in their profile. they should be removed
+     1 user that is linked and has active TCGA access. they should remain in the dbGapAuthorizedUsers group
+   */
+  it should "return NoContent and properly sync the whitelist for users of different link statuses" in withDefaultApiServices { services =>
+    Post("/sync_whitelist") ~> sealRoute(services.syncRoute) ~> check {
+      status should equal(NoContent)
+    }
+    services.rawlsDao.groups(dbGapAuthorizedUsersGroupName) should contain theSameElementsAs Set(services.thurloeDao.linkedUserId)
```

Comment:
is there reason to distrust `theSameElementsAs`? Given that it is a singleton set you could just as well test equality